### PR TITLE
docs(faq): more information on what changed in Helm 3

### LINKED ISSUE #5397: Incorporate changes made in Helm 3 into the documentation
Many features have been introduced into Helm 3, but the documentation has been lagging a bit behind. We should take this opportunity to re-write the documentation to capture the new architecture and to provide a landing spot for contributors writing new features/breaking changes in Helm 3.

### PR DESCRIPTION
**Pull request by bacongobbler**
builds upon the blog posts recently written about some of the changes in Helm 3. This also captures some major features that were identified in #5695.

collides with #5582 (cc @hickeyma)
closes #5397

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -10,12 +10,155 @@ send us a pull request.
...
+to completely remove Tiller.
+
+With Tiller gone, the security model for Helm is radically simplified. Helm 3 now supports all the modern security,
+identity, and authorization features of modern Kubernetes. Helm’s permissions are evaluated using your [kubeconfig file](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/).
+Cluster administrators can restrict user permissions at whatever granularity they see fit. Releases are still recorded
+in-cluster, and the rest of Helm’s functionality remains.
+
+### Release Names are now scoped to the Namespace
+
+With the removal of Tiller, The information Tiller used to store about each release had to go somewhere. In Helm 2, this
```
Comment (tariq1890) [inline docs/faq.md]:
```suggestion
With the removal of Tiller, the information it used to store about each release had to go somewhere. In Helm 2, this
```

File: docs/faq.md [hunk 2]
```diff
@@ -10,12 +10,155 @@ send us a pull request.
...
+  tags:
+    - database
+```
+
+Charts are still downloaded and placed in the charts/ directory, so subcharts vendored into the charts/ directory will continue to work without modification.
+
+### Name (or --generate-name) is now required on install
+
+In Helm 2, if no name was provided, an auto-generated name would be given. In production, this proved to be more of a
+nuisance than a helpful feature, so we now make Helm error out if no name is provided.
```
Comment (tariq1890) [inline docs/faq.md]:
```suggestion
nuisance than a helpful feature, so we now Helm will throw an error if no name is provided.
```

File: docs/faq.md [hunk 3]
```diff
@@ -10,12 +10,155 @@ send us a pull request.
...
+
+### Removal of helm serve
+
+`helm serve` ran a local Chart Repository on your machine for development purposes. However, it didn't receive much
+uptake as a development tool and had numerous issues with its design. In the end, we decided to remove it and split it
+out as a plugin.
+
+### Library chart support
+
+Helm 3 supports a class of chart called a “library chart”. This is a chart that is shared by other charts, but does not
```
Comment (tariq1890) [inline docs/faq.md]:
```suggestion
Helm 3 supports a class of charts called a “library chart”. This is a chart that is shared by other charts, but does not
```

File: docs/faq.md [hunk 4]
```diff
@@ -10,12 +10,155 @@ send us a pull request.
...
+create any release artifacts of its own. A library chart’s templates can only declare define elements. Globally scoped
```
Comment (tariq1890) [inline docs/faq.md]:
```suggestion
create any release artifacts of its own. A library chart’s templates can only declare and define elements. Globally scoped
```

File: docs/faq.md [hunk 5]
```diff
@@ -10,12 +10,155 @@ send us a pull request.
...
+non-define content is simply ignored. This allows users to re-use and share snippets of code that can be re-used across
```
Comment (tariq1890) [inline docs/faq.md]: Not sure about this but ....
```suggestion
undefined content is simply ignored. This allows users to re-use and share snippets of code that can be re-used across
```

File: docs/faq.md [hunk 6]
```diff
@@ -25,11 +168,12 @@ In Helm 2, in order to purge the release ledger, the `--purge` flag had to be pr
 functionality is now enabled by default. To retain the previous behaviour, use
 `helm uninstall --keep-history`.
 
-### Capabilities
+Additionally, several other commands were re-named to accomodate the same conventions:
```
Comment (tariq1890) [inline docs/faq.md]:
```suggestion
Additionally, several other commands were re-named to accommodate the same conventions:
```

File: docs/faq.md [hunk 7]
```diff
@@ -25,11 +168,12 @@ In Helm 2, in order to purge the release ledger, the `--purge` flag had to be pr
...
 
-Capabilities built-in has been simplified.
+- `helm inspect` -> `helm show`
+- `helm fetch` -> `helm pull`
 
-[Built-in Objects](chart_template_guide/builtin_objects.md)
+These commands also retained their older verbs as aliases, so you can continue to use them in either form.
```
Comment (tariq1890) [inline docs/faq.md]:
```suggestion
These commands have also retained their older verbs as aliases, so you can continue to use them in either form.
```

**Comment by tariq1890** [comment]
This PR was a helpful read. Thought I'd point these minor nits out which I found as I read through the whole diff.

File: docs/faq.md (hunk 3 above)
Comment (bacongobbler) [reply docs/faq.md]: This is proper grammar.

**Review by mattfarina — APPROVED** [review]

**Pull request merged**